src/core/meshcop/dtls.cpp

```diff
@@ -511,15 +815,73 @@ otError Dtls::MapError(int rval)
 
     switch (rval)
     {
+#if OPENTHREAD_ENABLE_APPLICATION_COAP_SECURE
+#ifdef MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
+    case MBEDTLS_ERR_PK_TYPE_MISMATCH:
+    case MBEDTLS_ERR_PK_FILE_IO_ERROR:
+    case MBEDTLS_ERR_PK_KEY_INVALID_VERSION:
+    case MBEDTLS_ERR_PK_KEY_INVALID_FORMAT:
+    case MBEDTLS_ERR_PK_UNKNOWN_PK_ALG:
+    case MBEDTLS_ERR_PK_PASSWORD_REQUIRED:
+    case MBEDTLS_ERR_PK_PASSWORD_MISMATCH:
+    case MBEDTLS_ERR_PK_INVALID_PUBKEY:
+    case MBEDTLS_ERR_PK_INVALID_ALG:
+    case MBEDTLS_ERR_PK_UNKNOWN_NAMED_CURVE:
+    case MBEDTLS_ERR_PK_BAD_INPUT_DATA:
+    case MBEDTLS_ERR_X509_SIG_MISMATCH:
+    case MBEDTLS_ERR_X509_BAD_INPUT_DATA:
+    case MBEDTLS_ERR_X509_FILE_IO_ERROR:
+    case MBEDTLS_ERR_X509_CERT_UNKNOWN_FORMAT:
+    case MBEDTLS_ERR_X509_INVALID_VERSION:
+    case MBEDTLS_ERR_X509_UNKNOWN_SIG_ALG:
+    case MBEDTLS_ERR_X509_INVALID_SERIAL:
+    case MBEDTLS_ERR_X509_UNKNOWN_OID:
+    case MBEDTLS_ERR_X509_INVALID_FORMAT:
+    case MBEDTLS_ERR_X509_INVALID_ALG:
+    case MBEDTLS_ERR_X509_INVALID_NAME:
+    case MBEDTLS_ERR_X509_INVALID_DATE:
+    case MBEDTLS_ERR_X509_INVALID_SIGNATURE:
+    case MBEDTLS_ERR_X509_INVALID_EXTENSIONS:
+    case MBEDTLS_ERR_X509_UNKNOWN_VERSION:
+#endif // MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
+    case -9186:
```

Comment:
Magic numbers: are there defines for these values?

Reply:
Forgotten values after a test with mbedtls. For these values no defines exist. Removed. Done.